src/wallet/rpcwallet.cpp

```diff
@@ -4282,6 +4315,19 @@ UniValue z_mergetoaddress(const UniValue& params, bool fHelp)
         max_tx_size = MAX_TX_SIZE_BEFORE_SAPLING;
     }
 
+    // This RPC does not support Sapling yet.
+    if (isToSaplingZaddr || containsSaplingZaddrSource) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, Sapling is not supported yet by z_mergetoadress");
+    }
+
+    // If this RPC does support Sapling...
+    // If Sapling is not active, do not allow sending from or sending to Sapling addresses.
+    if (!NetworkUpgradeActive(nextBlockHeight, Params().GetConsensus(), Consensus::UPGRADE_SAPLING)) {
+        if (isToSaplingZaddr || containsSaplingZaddrSource) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, Sapling has not activated");
```

Comment:
It's not possible to hit this code since we have already thrown an exception, but this is good defense if `z_mergetoaddress` gets sapling support before activation.